docker/archive/dest.go

```diff
@@ -19,17 +19,32 @@ func newImageDestination(ctx *types.SystemContext, ref archiveReference) (types.
 	if ref.destinationRef == nil {
 		return nil, errors.Errorf("docker-archive: destination reference not supplied (must be of form <path>:<reference:tag>)")
 	}
-	fh, err := os.OpenFile(ref.path, os.O_WRONLY|os.O_EXCL|os.O_CREATE, 0644)
+	// tries to open a file first to accomodate for load when using stdout
+	// as the file is created first then written to
+	// if the file doesn't exist then it creates a new file for exclusive use
+	// if the file exists, it checks to ensure the size is 0
+	fh, err := os.OpenFile(ref.path, os.O_WRONLY, 0644)
 	if err != nil {
-		// FIXME: It should be possible to modify archives, but the only really
-		//        sane way of doing it is to create a copy of the image, modify
-		//        it and then do a rename(2).
-		if os.IsExist(err) {
-			err = errors.New("docker-archive doesn't support modifying existing images")
+		fh, err = os.OpenFile(ref.path, os.O_WRONLY|os.O_EXCL|os.O_CREATE, 0644)
+		if err != nil {
+			// FIXME: It should be possible to modify archives, but the only really
+			//        sane way of doing it is to create a copy of the image, modify
+			//        it and then do a rename(2).
+			if os.IsExist(err) {
+				err = errors.New("docker-archive doesn't support modifying existing images")
+			}
+			return nil, err
+		}
+	} else {
+		fhStat, err := fh.Stat()
+		if err != nil {
+			return nil, err
 		}
-		return nil, err
-	}
 
+		if fhStat.Size() > 0 {
```

Comment:
Awkwardly, AFAICT it is not actually documented anywhere that `struct stat.st_size` has any particular value for pipes and FIFOs. I guess this could explicitly check for `(fhStat.Mode()&ModeType) == ModeNamedPipe || (fhStat.Mode().IsRegular() && fhStat.Size() == 0)` or something like that.